src/main/kotlin/nl/connectplay/scoreplay/controllers/UserController.kt

```diff
@@ -195,4 +196,32 @@ class UserController(
             call.respond(HttpStatusCode.InternalServerError)
         }
     }
+
+    suspend fun handleUpdateUserAsync(call: ApplicationCall) { // the call: Application is a small package that holeds the request and respons
+        val userId = call.parameters["id"]?.toIntOrNull()
+            ?: return call.respond(HttpStatusCode.BadRequest, "User ID is not a number")// read the id and cover it to int if possible if null badrequest
+        val updateDto = call.receiveNullable<UserUpdateDto>()
+            ?: return call.respond(HttpStatusCode.BadRequest, "Invalid or no update date") // receiveNullable checks if the UserUpdateDto is valid
+
+        try {
+            userRepository.updateUserAsync(userId, updateDto) // send the update to the UserRepository
```

Comment:
hier ook, we zien een directe call naar de user repository waar het vrij duidelijk is uit de woorden die je leest dat je een update uitvoert